data-prepper-plugins/s3-source/src/test/java/org/opensearch/dataprepper/plugins/source/S3SelectResponseHandlerTest.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source;
+
+import org.junit.Test;
+import software.amazon.awssdk.core.async.SdkPublisher;
+import software.amazon.awssdk.services.s3.model.SelectObjectContentEventStream;
+import software.amazon.awssdk.services.s3.model.SelectObjectContentResponse;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.mock;
+
+public class S3SelectResponseHandlerTest {
+    @Test
+    public void testS3SelectResponseHandlerWithMockedResponse() throws IOException {
+        S3SelectResponseHandler responseHandler = new S3SelectResponseHandler();
+        final SdkPublisher publisher = mock(SdkPublisher.class);
+        responseHandler.onEventStream(publisher);
+        responseHandler.responseReceived(mock(SelectObjectContentResponse.class));
+        responseHandler.complete();
+        final List<SelectObjectContentEventStream> receivedEvents =
+                responseHandler.getReceivedEvents();
+        assertNotNull(receivedEvents);
```

Comment:
This test should be able to verify the actual `receivedEvents` against a known input from the `responseReceived` call. Please update the test. Include some known input and check against that as expected output.

Reply:
Thanks David for your review, Changes are incorporated and available for review.